yacl/utils/circuit_executor.cc

```diff
@@ -82,6 +104,44 @@ void PlainExecutor<T>::Exec() {
   }
 }
 
+// template <typename T>
+// void PlainExecutor<T>::Finalize(absl::Span<T> outputs) {
+//   YACL_ENFORCE(outputs.size() >= circ_->nov);
+
+//   size_t index = wires_.size();
+//   for (size_t i = 0; i < circ_->nov; ++i) {
+//     dynamic_bitset<T> result(circ_->now[i]);
+//     for (size_t j = 0; j < circ_->now[i]; ++j) {
+//       result[j] = wires_[index - circ_->now[i] +
+//                          j];  // 得到的是逆序的二进制值 对应的混淆电路计算为
+//                               // LSB ^ d  输出线路在后xx位
+//     }
+//     outputs[circ_->nov - i - 1] = *(uint8_t*)result.data();
+//     // outputs[circ_->nov - i - 1] = *(uint128_t*)result.data();
+//     index -= circ_->now[i];
+//   }
+// }
+
+// template <typename T>
+// void PlainExecutor<T>::Finalize(absl::Span<T> outputs) {
+//   // YACL_ENFORCE(outputs.size() >= circ_->nov);
+
+//   size_t index = wires_.size();
+
+//   for (size_t i = 0; i < 32; ++i) {
+//     dynamic_bitset<T> result(8);
+//     for (size_t j = 0; j < 8; ++j) {
+//       result[j] =
+//           wires_[index - 8 + j];  // 得到的是逆序的二进制值
+//           对应的混淆电路计算为
+//                                   // LSB ^ d  输出线路在后xx位
+//     }
+//     outputs[32 - i - 1] = *(uint8_t*)result.data();
+//     // outputs[circ_->nov - i - 1] = *(uint128_t*)result.data();
+//     index -= 8;
+//   }
+// }
+
 template <typename T>
 void PlainExecutor<T>::Finalize(absl::Span<T> outputs) {
```

Comment:
Also, you can define a new `Finalize` method like `SetupInputs `; it accepts a pointer and a counter.